pkg/agent/storage/azure.go

```diff
@@ -0,0 +1,217 @@
+	blobs, err := a.Downloader.GetAllObjects()
+	if err != nil {
+		return fmt.Errorf("failed to get blobs: %v", err)
+	}
+	if len(blobs) == 0 {
+		return fmt.Errorf("no blobs found")
+	}
+	if err := a.Downloader.Download(blobs); err != nil {
+		return fmt.Errorf("failed to download blobs: %v", err)
+	}
+	return nil
+}
+
+func newDefaultDownloader(serviceUrl string) (AzureDownloader, error) {
+	// Initialize AzureBlobClient
+	var err error
+	var client *azblob.Client
+	if _, ok := os.LookupEnv(azure.AzureClientSecret); ok {
+		// Load Azure Credentials
+		credential, err := azidentity.NewDefaultAzureCredential(nil)
+		if err != nil {
+			return nil, fmt.Errorf("failed to load Azure credentials: %v", err)
+		}
+		client, err = azblob.NewClient(serviceUrl, credential, nil)
+		if err != nil {
+			return nil, fmt.Errorf("failed to create Azure client with credentials: %v", err)
+		}
+	} else {
+		client, err = azblob.NewClientWithNoCredential(serviceUrl, nil)
+		if err != nil {
+			return nil, fmt.Errorf("failed to create Azure client with no credentials: %v", err)
+		}
+	}
+
+	return &defaultAzureDownloader{
+		client: client,
+	}, nil
+}
+
+// ParseAzureUri is a method that parses the Azure URI and sets the azureUriParts.
+// The Azure URI should follow this specific format: azure://<serviceUrl>/<containerName>/<blob>
+//
+// Here's a breakdown of the URI components:
+//   - <serviceUrl>: The base URL of your Azure storage account. For example, 'myStorageAccount.blob.core.windows.net'
+//   - <containerName>: The specific containerName in the Azure storage account. For example, 'myContainer'
+//   - <blob>: The specific blob within the containerName. This can include a path with directories. For example, 'this/is/virtualDir/fileName'
+//
+// Putting it all together, an example Azure storageURI would look like this:
+// 'azure://myStorageAccount.blob.core.windows.net/myContainer/this/is/virtualDir/fileName'
+//
+// In this example,
+//   - serviceUrl would be 'myStorageAccount.blob.core.windows.net'
+//   - containerName would be 'myContainer'
+//   - prefix (the path to the blob within the containerName) would be 'this/is/virtualDir/fileName'.
+func parseAzureUri(uri string) (azureUriParts, error) {
+	if !strings.HasPrefix(uri, string(AZURE)) {
+		return azureUriParts{}, fmt.Errorf("invalid Azure URI: '%s'. The URI must start with '%s'", uri, string(AZURE))
+	}
+
+	uri = strings.TrimPrefix(uri, string(AZURE))
+	tokens := strings.SplitN(uri, "/", 3)
+	if len(tokens) < 3 {
+		return azureUriParts{}, fmt.Errorf("invalid Azure URI: '%s'. The URI must be in the format 'azure://<serviceUrl>/<containerName>/<blob>'", uri)
+	}
+
+	serviceUrl := tokens[0]
+	containerName := tokens[1]
+	prefix := ""
+	if len(tokens) == 3 {
+		prefix = tokens[2]
+	}
+
+	return azureUriParts{
+		serviceUrl:    string(HTTPS) + serviceUrl,
+		containerName: containerName,
+		prefix:        prefix,
```

Comment:
better to name it as blob to improve the readability.

Reply:
Thank you for the suggestion. The initially confusing name might suggest that it is a single blob, but it is actually a virtual directory that contains one or more blobs. I have updated the variable name to better reflect its true purpose in commit c143385f0e2371c5e121c58b27ebb26d6abf9aab